google-cloud-translate-v2/lib/google/cloud/translate/v2/service.rb

```diff
@@ -118,7 +118,9 @@ def default_http_headers
                 "Content-Type"                 => "application/json",
                 "x-goog-api-client"            => "gl-ruby/#{RUBY_VERSION} gccl/#{lib_version}"
               }
-              headers["x-goog-user-project"] = credentials.quota_project_id if credentials.respond_to? :quota_project_id
+              quota_project = V2.translation_config&.quota_project
+              quota_project ||= credentials.quota_project_id if credentials.respond_to? :quota_project_id
```

Comment:
I'm having trouble following how this existing code works. What type of credentials has `quota_project_id`?

Reply:
`quota_project` is the ID of the project against which to charge quota, which could be different from the project that hosts the resources being manipulated. It needs to be set in the "x-goog-user-project" header. The source of the data is normally a field in the credentials, but it can be overridden via configuration.

The generated libraries (both gapic and apiary) all do this already, but since Translate V2 is completely handwritten, it needs to have manual code that does this.